oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -218,6 +226,9 @@ private ImpersonatedCredentials(Builder builder) {
     if (this.scopes == null) {
       throw new IllegalStateException(SCOPE_EMPTY_ERROR);
     }
+    if (this.lifetime > TWELVE_HOURS_IN_SECONDS) {
```

Comment:
While checking to see if this could possibly be null (it can't but could be `0`), it looks like we can possibly make a request for a token with `0s` for the lifetime value. Should it actually be unset and rely on the server default value?

Reply:
Filed https://github.com/googleapis/google-auth-library-java/issues/526